### PR TITLE
fix: Support TBC classic

### PR DESCRIPTION
**Pull request by Nessworthy**
Commit contains changes required for this addon to work with TBC classic's addon API changes

### CODE REVIEW

**Comment by darfink** [comment]
That's awesome! I've stopped playing WoW: Classic since I released this plugin, but I still have a soft spot for it.

As of now, I don't have the ability to test these changes. Is there any possibility you are able to confirm whether these are all changes required to ensure the plugin is TBC compatible?

I'd happily merge the PR in its current state, but I've left some input.

File: Source/ItemDatabase.lua [hunk 1]
```diff
@@ -7,8 +7,8 @@ local FuzzyMatcher = require 'Utility.FuzzyMatcher'
 -- Consts
 local const = util.ReadOnly({
   -- Find highest ID @ https://classic.wowhead.com/items?filter=151;2;24283
-  disjunctItemIds = {172070},
-  highestItemId = 24283,
+  disjunctItemIds = {172070, 12270, 122284, 180089},
```
Comment (darfink) [inline Source/ItemDatabase.lua:10]: The `12270` ID is already covered by the first range in `itemIdRanges`. Is the ID redundant or a typo?

File: Source/ChatAutocompleteIntegrator.lua [hunk 2]
```diff
@@ -130,9 +131,9 @@ function ChatAutocompleteIntegrator:_OnItemSearchComplete(editBox, items, search
 
   -- Since this is received asynchronously, discard the result if it has become irrelevant
   if util.IsNilOrEmpty(searchTerm) or searchTerm:find(searchInfo.searchTerm, nil, true) ~= 1 then
+    util.PrettyPrint('Item search DISCARDED' + searchTerm)
```
Comment (darfink) [inline Source/ChatAutocompleteIntegrator.lua:134]: Is this print a left-over from debugging or does it fulfill another purpose?

File: Source/ItemDatabase.lua (hunk 1 above)
Comment (Nessworthy) [reply Source/ItemDatabase.lua:10]: Most likely a typo, I'll go back and have a second pass :D

File: Source/ChatAutocompleteIntegrator.lua (hunk 2 above)
Comment (Nessworthy) [reply Source/ChatAutocompleteIntegrator.lua:134]: Most likely left over debugging from trying to trigger this.

**Comment by Nessworthy** [comment]
Your addon page shows there's a settings window somewhere, how would I go about toggling this?
I think that's the last thing remaining, although there is a bit of a search slowdown due to all the new items.

Thanks for the plugin by the way, really neat!

**Comment by darfink** [comment]
From my understanding, TBC: Classic and Vanilla Classic will remain as two separate environments, each having their own community of plugins. Correct me if I'm wrong, but if that is the case, is a toggle required? Shouldn't the TBC & vanilla implementations be their own separate plugin versions? Or were you referring to a toggle for something else? I'd assume, for example, that changing the `Interface` version would render this plugin _out of date_ on a vanilla client?

Regarding the slowdown, and as you described, must have impaired the search significantly. It's almost twice the amount of items to filter. Sadly, I've already tried to optimize the search implementation, but traditional computer science have not seen to be applicable with Lua. I'd guess its due to it being such a high-level language, or at least I haven't achieved any performance improvements by using data structures optimal for the task. With that said, there are definitely optimizations available with some black magic and Lua expertise.

If you are able, you could try increasing the `itemsSearchedPerUpdate` constant to `2000`. This might introduce slight stuttering when searching for items in-game, but at least the search results will be provided twice as fast. It might be worth the trade-off.

I'm glad you appreciate it! I was inspired by [fzf](https://github.com/junegunn/fzf) and its fantastic fuzzy search, which made me much more productive in the shell, and I couldn't help wonder how awesome such a fuzzy search would be in WoW 🙂

**Comment by darfink** [comment]
Thanks a ton for the pull request, I'll try to make the final touch ups myself. If you are able to tell whether the ID `12270` should be replaced with something else, please do tell.

**Comment by Nessworthy** [comment]
Should be `122270`!

**Pull request merged**